### PR TITLE
fix(api): auto-reconcile v2 boxes stuck in ERROR back toward STARTED

### PR DESCRIPTION
**Pull request by G4614**
while v2 box running inside runner but marked as "Error" by mistake by API, scan reconcile them each minute

## Test plan

- [x] `yarn nx test api --testPathPatterns=box.manager.reconcile --skip-nx-cache`
- [x] `yarn nx test api --testPathPatterns=job-state-handler.service --skip-nx-cache`
- [x] `yarn tsc -p api/tsconfig.json --noEmit`
- [x] `yarn eslint api/src/box/managers/box.manager.ts api/src/box/managers/box.manager.reconcile.spec.ts api/src/box/services/job-state-handler.service.ts api/src/box/services/job-state-handler.service.spec.ts`

## Summary by CodeRabbit — Release Notes

* **New Features**
  * Added automatic reconciliation for boxes stuck in error states, including a scheduled, lock-protected workflow with retry-attempt ceilings.
  * Added logic to better handle split-brain “box already exists” collisions by converging boxes to a recovered state instead of leaving them in error.
  * Added job-backed failure accounting so retry limits work even when expected job records aren’t created.

* **Bug Fixes**
  * Prevented retries for boxes that hit the configured recovery-attempt cap; skipped incompatible runner versions and already-locked boxes.

* **Tests**
  * Added Jest coverage for reconciliation and job failure handling edge cases.

### CODE REVIEW

**Comment by nieyy** [comment]
One concern after looking at the recovery contract a bit more: this PR makes `box_id` an explicit argv element for the shim, which is the right direction, but the consumer side still validates with `cmdline.contains(box_id)` in `is_same_process_linux`.

That part is pre-existing, so this is not something the PR newly broke. But since this PR is formalizing the “PID belongs to this box” contract, I think it would be safer to tighten the check here as well: split `/proc/<pid>/cmdline` into argv entries and require one arg to exactly equal `box_id`, instead of doing a substring search over the whole cmdline.

The edge case is something like querying `box-1` while another live shim has `box-10` in argv. With `contains`, that can still look like a match. Since `shim_argv()` now gives us a clean standalone `box_id` arg, exact argv matching seems like the contract we actually want. A small regression test for the prefix/substr case would make this much harder to regress later.

**Comment by nieyy** [comment]
One more small test-contract point: the new `test_is_same_process_accepts_live_shim_with_box_id_in_argv` says it rejects “no false positive from a coincidental substring”, but the negative case uses `totally-different-id`, which does not actually exercise the substring/prefix problem.

If we keep the exact-match tightening above, it would be nice to make the test prove that case directly. For example, spawn the stand-in with something like `box-10` in argv, then assert that querying `box-1` returns false. That is the case `cmdline.contains(box_id)` would get wrong, and it would make the test match the contract it is trying to pin down.

**Comment by DorianZheng** [comment]
Duplicate in #590

File: apps/api/src/box/managers/box.manager.ts [hunk 1]
```diff
@@ -289,6 +306,120 @@ export class BoxManager implements TrackableJobExecutions, OnApplicationShutdown
...
+            resourceType: ResourceType.BOX,
+            resourceId: candidate.id,
+            type: JobType.START_BOX,
+            status: JobStatus.FAILED,
+            createdAt: MoreThan(failureWindowStart),
+          },
+        })
+        if (recentFailures >= MAX_RECOVER_ATTEMPTS) {
+          continue
+        }
```
[the remark this answers is not in the document]
Comment (G4614) [reply apps/api/src/box/managers/box.manager.ts:383]: Fixed in c227f92b.

`syncInstanceState`'s catch block now records a terminal **FAILED START_BOX** job before transitioning the box to ERROR — gated to a STARTED desired state with a known runner, and best-effort so it never blocks the ERROR transition. `completedAt` is set so the synthetic row stays outside the `IDX_UNIQUE_INCOMPLETE_JOB` partial-unique index (`WHERE "completedAt" IS NULL`).

So the synchronous-failure path (runner lookup / adapter creation throwing before `runnerAdapter.startBox()` persists its job) now produces a FAILED START_BOX record, and the reconcile ceiling counts it instead of re-driving the box out of ERROR forever. Async START_BOX failures are handled by `job-state-handler` (not this catch), so there's no double counting.

Added `box.manager.sync-error-job.spec.ts`: a STARTED sync that throws records exactly one FAILED START_BOX job; a STOPPED sync and a runner-less box record none. Verified reproduce-before-fix (reverting the fix fails the STARTED case).
